docs/site/migration/models/operation-hooks.md

```diff
@@ -6,8 +6,38 @@ sidebar: lb4_sidebar
 permalink: /doc/en/lb4/migration-models-operation-hooks.html
 ---
 
-{% include note.html content="
-This is a placeholder page, the task of adding content is tracked by the
-following GitHub issue:
-[loopback-next#3952](https://github.com/strongloop/loopback-next/issues/3952)
-" %}
+Operation hooks are not supported in LoopBack 4 yet. It will be, eventually; you
+can follow [this issue](https://github.com/strongloop/loopback-next/issues/1919)
+to keep track of the progress being made.
+
+In the meantime, we are providing a temporary API for enabling Operation hooks
+in LoopBack 4: override `DefaultCrudRepository`'s `definePersistedModel` method
+in the model's repository.
+
+The `definePersistedModel` method of `DefaultCrudRepository` returns a model
+instance on which you can apply the
+[LoopBack 3 operation hooks](https://loopback.io/doc/en/lb3/Operation-hooks.html).
+Make sure to return the model instance from the derived class.
+
+Here is an example of a repository implementing `definePersistedModel` and
+applying an operation hook on a model instance:
+
+```ts
+class ProductRepository extends DefaultCrudRepository<
+  Product,
+  typeof Product.prototype.id,
+  ProductRelations
+> {
+  constructor(dataSource: juggler.DataSource) {
+    super(Product, dataSource);
+  }
+
+  definePersistedModel(entityClass: typeof Product) {
+    const model = super.definePersistedModel(entityClass);
+    model.observe('before save', async ctx => {
```

Comment:
We can access the model as `this.modelClass`. Why not leveraging that to set up observers instead of overriding `definedPersistedModel`?

Reply:
Initially we were considering having an `observe` method on `DefaultCrudRepository`, which would call `this.modelClass.observe`. After some brainstorming, @bajtos suggested the possible problems with `observe` is not worth it. Eg: calling `observe()` multiple times and how to handle it. He had suggested another approach but it wasn't very DX-friendly, so we settled on the current one.

"You want to directly access the model? Then you will have to override the `definePersistedModel ` method."

Reply:
@raymondfeng please check the description of the user story and the discussion in that GH issue for more context.

@hacksparrow please update PR description with a reference to the GH issue this is related to.

Reply:
@hacksparrow We do support multiple observers for a given model class/operation - see https://github.com/strongloop/loopback-datasource-juggler/blob/master/lib/observer.js#L51

Reply:
Yes, that's understood. We were wondering how should LB4 respond when an operation hook is installed multiple times: a. Throw an error? b. Accept silently - how does the user know what happened?

Also, it is not impossible to call the `observe` method before the `modelClass` is defined. What happens in that case?

Since this is a temporary workaround, we didn't want to invest too much time handling various possible scenarios, edge cases etc. We can do that when we investigate the use of interceptor for the "real" implementation.